extension/js/common/ui/key-import-ui.ts

```diff
@@ -105,18 +105,18 @@ export class KeyImportUi {
     const attach = new AttachmentUI(() => Promise.resolve({ count: 100, size: 1024 * 1024, size_mb: 1 }));
     attach.initAttDialog('fineuploader', 'fineuploader_button', {
       attAdded: async file => {
-        let prv: OpenPGP.key.Key | undefined;
+        let prv: Key | undefined;
         const utf = file.getData().toUtfStr();
         if (utf.includes(PgpArmor.headers('privateKey').begin)) {
           const firstPrv = MsgBlockParser.detectBlocks(utf).blocks.filter(b => b.type === 'privateKey')[0];
           if (firstPrv) { // filter out all content except for the first encountered private key (GPGKeychain compatibility)
-            prv = (await opgp.key.readArmored(firstPrv.content.toString())).keys[0];
+            prv = (await KeyUtil.parse(firstPrv.content.toString()));
```

Comment:
But now I'm thinking of this one :) Eventually we'll make `KeyUtil.parse` throw when it encounters more than one key.

As long as we do have a test for this scenario: `// filter out all content except for the first encountered private key (GPGKeychain compatibility)` then it's no problem - whenever we change it, we'll bump into the test and resolve it.

Actually, it's not the most ideal as is - if I give it two private keys, I may actually want to import two private keys. But now we're talking about importing more than one key at once, which is a different issue, I'll make a separate ticket for it.

Reply:
Yeah, I think at the end of the road we'll need to think about some clever way of merging all the `parse` functions but at the moment I'm working on handling private S/MIME keys. Once I've got that maybe I'll have a nice idea on how to join all these divergent ways of handling keys :)